relay/relay.go

```diff
@@ -328,6 +337,9 @@ func (r *Relay) join(ctx context.Context) {
 }
 
 func (r *Relay) addSignatureForPeer(peerID core.PeerID, signature []byte) {
+	r.mu.Lock()
```

Comment:
missing `mu` usage in the `AmITheLeader`,  `PeerCount`, `Clean` functions. 
You used this mutex for both peers field and signature field.